Cython/Utility/CppConvert.pyx

```diff
@@ -4,66 +4,47 @@
 #################### string.from_py ####################
 
 cdef extern from *:
-    cdef cppclass string "{{type}}":
-        string() except +
-        string(char* c_str, size_t size) except +
     cdef const char* __Pyx_PyObject_AsStringAndSize(object, Py_ssize_t*) except NULL
 
 @cname("{{cname}}")
-cdef string {{cname}}(object o) except *:
+cdef Tp {{cname}}(object o) except *:
     cdef Py_ssize_t length = 0
     cdef const char* data = __Pyx_PyObject_AsStringAndSize(o, &length)
-    return string(data, length)
+    return Tp(data, length)
 
 
 #################### string.to_py ####################
 
-#cimport cython
-#from libcpp.string cimport string
-cdef extern from *:
-    cdef cppclass string "{{type}}":
-        char* data()
-        size_t size()
-
 {{for py_type in ['PyObject', 'PyUnicode', 'PyStr', 'PyBytes', 'PyByteArray']}}
 cdef extern from *:
     cdef object __Pyx_{{py_type}}_FromStringAndSize(const char*, size_t)
 
 @cname("{{cname.replace("PyObject", py_type, 1)}}")
-cdef inline object {{cname.replace("PyObject", py_type, 1)}}(const string& s):
+cdef inline object {{cname.replace("PyObject", py_type, 1)}}(const Tp& s):
     return __Pyx_{{py_type}}_FromStringAndSize(s.data(), s.size())
 {{endfor}}
 
 
 #################### vector.from_py ####################
 
-cdef extern from *:
-    cdef cppclass vector "std::vector" [T]:
-        void push_back(T&) except +
-
 @cname("{{cname}}")
-cdef vector[X] {{cname}}(object o) except *:
-    cdef vector[X] v
+cdef Tp {{cname}}(object o) except *:
```

Comment:
Did you really mean to use `Tp` as a type name here? Does that work?

Reply:
Yes it seems to work. It's extracted from the context in exactly the same way that `X` used to be.